js/ui/components/clubhouse.js

```diff
@@ -155,6 +155,50 @@ var ClubhouseWindowTracker = new Lang.Class({
     },
 });
 
+var ClubhouseAnimation = new Lang.Class({
+    Name: 'ClubhouseAnimation',
+    Extends: Animation,
```

Comment:
I had mentioned you don't need to extend this and instead just give it the already formatted frames. I see that to support the random frames (which AFAIK are not used in the Shell for now), overriding the `_getCurrentDelay` makes it easier.
So in the interest of not requiring yet more changes, I am okay with this. But I think we should change it soon so that we don't parse the format again in the Shell, i.e.:
- We keep this class, but feed it an array of objects like `{'index': 10, 'delay-interval': [2000, 5000]}`
(So, in future changes to the format we only change the interpretation in one place.)